spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryProperties.java

```diff
@@ -159,19 +164,36 @@
 
 	/**
 	 * Map of serviceId's -> tag to query for in server list. This allows filtering
-	 * services by a single tag.
+	 * services by a single tag. (Deprecated - instead use 'serviceInstanceQueryTags')
 	 */
+	@Deprecated
 	private Map<String, String> serverListQueryTags = new HashMap<>();
```

Comment:
So this property is not used internally by anything. If someone configures this - it does nothing. I deprecated it along w/ its associated default property `defaultQueryTag`. I am not sure the proper way to deprecate config props in Spring Cloud (in Boot there is the `@DeprecatedConfigurationProperty` but its not in use here).

Reply:
That's the way we deprecate it, however, please make sure you also add `deperecated` tag and link to the new version in the javadocs whenever you deprecate items.